Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraController.cs

```diff
@@ -0,0 +1,155 @@
+            viewInstance.TakeScreenshotButton.onClick.AddListener(RequestTakeScreenshot);
+            viewInstance.CameraReelButton.onClick.AddListener(OpenCameraReelGallery);
+            viewInstance.ShortcutsInfoButton.onClick.AddListener(ToggleShortcutsInfo);
+
+            shortcutsController = new ScreencaptureShortcutsController(() => viewInstance.ShortcutsInfoPanel);
+            mvcManager.RegisterController(shortcutsController);
+        }
+
+        public override void Dispose()
+        {
+            viewInstance!.CloseButton.onClick.RemoveListener(RequestDisableInWorldCamera);
+            viewInstance.TakeScreenshotButton.onClick.RemoveListener(RequestTakeScreenshot);
+            viewInstance.CameraReelButton.onClick.RemoveListener(OpenCameraReelGallery);
+            viewInstance.ShortcutsInfoButton.onClick.RemoveListener(ToggleShortcutsInfo);
+
+            base.Dispose();
+        }
+
+        public void Show()
+        {
+            sidebarButton.OnSelect(null);
+            mvcManager.ShowAsync(InWorldCameraController.IssueCommand(new ControllerNoData()));
+
+            bool hasSpace = storageService.StorageStatus.HasFreeSpace;
+            viewInstance!.TakeScreenshotButton.gameObject.SetActive(hasSpace);
+            viewInstance.NoStorageNotification.gameObject.SetActive(!hasSpace);
+        }
+
+        public void Hide(bool isInstant = false)
+        {
+            sidebarButton.OnDeselect(null);
+            viewInstance.HideAsync(default(CancellationToken), isInstant).Forget();
+        }
+
+        protected override UniTask WaitForCloseIntentAsync(CancellationToken ct) =>
+            UniTask.WhenAny(
+                viewInstance!.CloseButton.OnClickAsync(ct),
+                viewInstance.CameraReelButton.OnClickAsync(ct));
+
+        public void PlayScreenshotFX(Texture2D image, float splashDuration, float middlePauseDuration, float transitionDuration)
+        {
+            UIAudioEventsBus.Instance.SendPlayAudioEvent(viewInstance.SFXScreenshotCapture);
+            viewInstance.ScreenshotCaptureAnimation(image, splashDuration, middlePauseDuration, transitionDuration);
+        }
+
+        private void OpenCameraReelGallery()
+        {
+            RequestDisableInWorldCamera();
+
+            mvcManager.ShowAsync(
+                ExplorePanelController.IssueCommand(new ExplorePanelParameter(ExploreSections.CameraReel, BackpackSections.Avatar)));
+        }
+
+        private void RequestTakeScreenshot()
+        {
+            if (!world.Has<TakeScreenshotUIRequest>(camera!.Value))
+                world.Add(camera!.Value, new TakeScreenshotUIRequest { Source = "UI" });
+        }
+
+        private void RequestDisableInWorldCamera()
+        {
+            if (CameraIsActivated())
+                world.Add(camera!.Value, new ToggleInWorldCameraUIRequest { IsEnable = false });
+
+            bool CameraIsActivated() =>
```

Comment:
Wouldm't it be better to move the conditions into the IF and add a comment if some clarification is needed?

Reply:
I again here on Robert Martin side of  _"The proper use of comments is to compensate for our failure to express ourselves in code."_ and Donald Knuth _"Instead of commenting your code, write your code in such a way that it explains itself."_ . There are known issues with comments that just describes "What (it is)", most known that comment easily get outdated, but also brings noise. 

So generally, there are 3 main rules for comments:
1. Explain Why, Not What
2. Provide Context (like for workarounds, hacks)
3. Document Non-Obvious Logic (for difficult algorithmic, computational and so on)

otherwise I prefer extracting to Methods with good names

Reply:
I don't agree with that way of thinking, but this is not an important thing and there is no internal rule for it so your preference is ok.